improver_tests/calibration/ensemble_calibration/test_ApplyEMOS.py

```diff
@@ -0,0 +1,170 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the `ensemble_calibration.ApplyEMOS` class."""
+
+import unittest
+
+import iris
+import numpy as np
+from iris.tests import IrisTest
+
+from improver.calibration.ensemble_calibration import ApplyEMOS
+from improver.utilities.cube_manipulation import get_dim_coord_names
+
+from ...set_up_test_cubes import (
+    set_up_percentile_cube, set_up_probability_cube, set_up_variable_cube)
+
+
+def build_coefficients_cube(data, template):
+    """Make a cube of coefficients with expected metadata"""
+    index = iris.coords.Coord(np.arange(4), long_name="coefficient_index")
+    name = iris.coords.Coord(["gamma", "delta", "alpha", "beta"],
+                             long_name="coefficient_name")
+    coefficients = iris.cube.Cube(data, long_name="emos_coefficients",
+                                  dim_coords_and_dims=[(index, 0)],
+                                  aux_coords_and_dims=[(name, 0)])
+
+    # add spatial and temporal coords from forecast to be calibrated
+    for coord in ["time", "forecast_period", "forecast_reference_time"]:
+        coefficients.add_aux_coord(template.coord(coord).copy())
+
+    for coord in [template.coord(axis='x'), template.coord(axis='y')]:
+        bounds = [min(coord.points), max(coord.points)]
+        point = np.median(bounds)
+        new_coord = coord.copy(points=[point], bounds=[bounds])
+        coefficients.add_aux_coord(new_coord)
+
+    coefficients.attributes['diagnostic_standard_name'] = 'air_temperature'
+
+    return coefficients
+
+
+class Test_process(IrisTest):
```

Comment:
OK so there is at least one ugly bug in land sea masked calibration, which manifests through an error raised when merging calibrated and uncalibrated points, that means this is not trivial.  So I am going to commit the new test I have so far, which fails due to a coordinate var name that is added at some point, probably within ECC.  I will then come back to this and see whether it is just percentiles that are impacted by this bug, add enough tests to cover all instances and fix it.  But I can't get this finished today I'm afraid.

Reply:
Just to try and help, I think you're probably referring to [this line](https://github.com/metoppv/improver/blob/master/improver/ensemble_copula_coupling/utilities.py#L167), where a percentile coordinate is created with a `var_name`.

Reply:
Thanks @gavinevans , that helps.  The correct fix for this (and how much other code needs to be changed) depends on the metadata standard - I've sent an email offline so we can discuss there.

Reply:
Contrary to my worst fears removing `var_name="percentile"` from this statement doesn't cause any problems!  So I've passed this back to you @gavinevans .  Will look at the wider `var_name` behaviour in a separate PR.

Reply:
As far as I can tell, one acceptance test (test_percentiles_input_land_sea) is still failing because of the change in `improver/ensemble_copula_coupling/utilities.py`. Would reverting that change in this PR, and putting it into #1224 instead, solve this issue?

Reply:
Sounds like a good idea cos the same test is failing in #1224

Reply:
Ah, this is complicated.  I can't **not** fix this in this PR, as it would cause unit test failures.  But evidently I can't fix it here without breaking tests elsewhere.

One way to deal with this would be for me to add the "strip var names" on load (just the changes from load.py) into this PR.  However that would break other ECC tests (for realization).

I think perhaps we should discuss.

Reply:
OK having had a chat with @gavinevans he's suggested I make this change in the other PR and we move that one along first.  Once #1224 has been merged the tests on this PR should all pass.  So I'll label this as "Blocked" for now.